code-challenges.test.js

```diff
@@ -13,7 +13,38 @@
 
 // --------------------1) Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
 
+// crate a function 
+// have it assign new variables that take in value and replaces it with a new value
+//return changed string
+
 // a) Create a test with expect statements using the variables provided.
+describe("hello", () => {
+
+    const secretCodeWord1 = "Lackadaisical"
+    const secretCodeWord2 = "Gobbledygook"
+    const secretCodeWord3 = "Eccentric"
+
+ 
+    it("returns a string that says hiCreate a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.", () => {
+  
+      
+      expect(no(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+      expect(no(secretCodeWord2)).toEqual("Gobbledygook")
+      expect(no(secretCodeWord3)).toEqual("Eccentric")
+    })
+  })
+
+
+
+
+  const no = (str) => {
+     a = str.replace(/[a]/gi, '4')
+     b = a.replace(/[e]/gi, '5')
+     c = b.replace(/[i]/gi, '1')
+     return c.replace(/[o]/gi, '0')
+  }
```

Comment:
Regex is pretty cool. I would encourage you to use your pseudo code as an opportunity to describe your function. In this case, explaining what the regex is doing.